### PR TITLE
Links to native system requirements (resolves #371)

### LINKED ISSUE #371: Document native package requirements
The README should mention all the native packages that we need, e.g. Azure now needs  libffi-dev. There may be others. Check CI build to verify. Then double-check on a bare-bones Ubuntu install.

### PR DESCRIPTION
**Pull request by cket**

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -1,6 +1,9 @@
 Toil is a scalable, efficient, cross-platform pipeline management system, written entirely in
 Python, and designed around the principles of functional programming. Full documentation can be found at `Read the Docs`_.
 
+Native system requirements: `python2.7-dev`_, make_, gcc_, `libffi-dev`_,
 .. _Read the Docs: http://toil.readthedocs.org/en/latest/
-
-
+.. _python2.7-dev: http://packages.ubuntu.com/search?keywords=python2.7-dev
```
Comment (cket) [inline README.rst]: This was the most official documentation for the python-dev package I could find, but it still isn't very informative.

File: README.rst [hunk 2]
```diff
@@ -1,6 +1,9 @@
 Toil is a scalable, efficient, cross-platform pipeline management system, written entirely in
 Python, and designed around the principles of functional programming. Full documentation can be found at `Read the Docs`_.
 
+Native system requirements: `python2.7-dev`_, make_, gcc_, `libffi-dev`_,
```
Comment (hannes-ucsc) [inline README.rst]: This should go into the installation section of the Sphinx doc. And it should be qualified that these are only needed for the encryption extra. I think we already have a section that describes each extra and the native requirements should be listed there.

Comment (hannes-ucsc) [inline README.rst]: And I would use generic names for these: "Python headers and static libraries", "Make", "GCC", and libffi library and headers. Then mention the steps that needs to be run to get these on Ubuntu and OS X.

File: docs/essentials.rst [hunk 3]
```diff
@@ -83,15 +87,29 @@ Here's what each extra provides:
 
 * The ``mesos`` extra provides support for running Toil on an `Apache Mesos`_
   cluster. Note that running Toil on SGE (GridEngine), Parasol or a single
-  machine is enabled by default and does not require an extra.
+  machine is enabled by default and does not require an extra. Requires the
+  `Python headers and static libraries`_ which can be installed on Ubuntu
+  by running 'apt-get install python-dev'. On OSX, these headers are installed
+  when you install the xcode command line tools by running 'xcode-select --install'
```
Comment (hannes-ucsc) [inline docs/essentials.rst]: OS X

Comment (hannes-ucsc) [inline docs/essentials.rst]: Xcode

File: docs/essentials.rst [hunk 4]
```diff
@@ -51,8 +51,12 @@ Prerequisites
 
 * Python 2.7.x
 
+* Make_
```
Comment (hannes-ucsc) [inline docs/essentials.rst]: Why does a pip installation need Make?

**Comment by hannes-ucsc** [comment]
Never mind. I'm taking this.

**Pull request merged**